PV260_Project/PresentationLayer/Utils/IConsoleWrapper.cs

```diff
@@ -0,0 +1,10 @@
+namespace PresentationLayer.Utils;
+
+/// <summary>
+/// Serves as a wrapper for System.Console IO methods, ReadLine and WriteLine, allowing them to be mocked.
+/// </summary>
+public interface IConsoleWrapper
```

Comment:
Good idea to wrap the communication through wrapper, I guess I would choose different name, but it is just my opinion :D

Reply:
Maybe to make it even robust, it does not have to be `Console` wrapper, but something more generic, so it can read and write input based on the implementation (in our case there can be only console implementation).

Reply:
How would you call it? Maybe I could create something like `IIoWrapper` with `Read` and `Write` methods, and `ConsoleIoWrapper` would simply use `System.Console.ReadLine` for `Read`, as it is now? (`System.Console.Read` also exists, but maybe for our use case it's fine - since I feel like `IIoWrapper` shouldn't have `ReadLine` / `WriteLine` method)

Reply:
Yea, the names of the generic one should have been something else for better understanding (maybe just `Read` and `Write`), but you dont have to change it, it was just an opinion :)

Reply:
I added `IIoWrapper` with generic methods `GetInput` and `ShowMessage`, which makes it more readable IMO, good point. (now there is an empty interface `IConsoleIoWrapper`, but that shouldn't be that big of an issue I believe)